core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -432,7 +432,7 @@ private[spark] class Executor(
           setTaskFinishedAndClearInterruptStatus()
           execBackend.statusUpdate(taskId, TaskState.KILLED, ser.serialize(TaskKilled(t.reason)))
 
-        case _: InterruptedException if task.reasonIfKilled.isDefined =>
+        case _: Throwable if task.reasonIfKilled.isDefined =>
```

Comment:
I think NonFatal is probably fine here -- an OOM could presumably cause these messages to be dropped and the task marked as failed anyways.